src/sensai/data_transformation.py

```diff
@@ -45,30 +51,27 @@ def _fit(self, df: pd.DataFrame):
     def _apply(self, df: pd.DataFrame) -> pd.DataFrame:
         pass
 
+    @trackDFHistory
```

Comment:
I was almost sure you would say it and after I finished I thought similarly :D. Will remove it, was just nice to write a decorator again

Reply:
It's gone. If didn't want to persist the infos together with the object instances, I could have made the decorator useful by persisting the stuff not in the instance but in some global dict. But I did not want to make it transient so I saw no good solution.

Just out of curiosity, do you have an idea how to make it work as decorator without this ugly hack I did?

Reply:
Because changing the functionality of the method is not the only effect you want (you also want side-effects on the object's state), this simply isn't a use case for a function decorator.